app/components/chart_component.rb

```diff
@@ -6,11 +6,11 @@ class ChartComponent < ViewComponent::Base
   renders_one :header
   renders_one :footer
 
-  attr_reader :school, :chart_type, :analysis_controls, :no_zoom, :axis_controls, :html_class, :fuel_type
+  attr_reader :school, :chart_type, :analysis_controls, :no_zoom, :axis_controls, :html_class, :fuel_type, :show_advice, :autoload_chart, :wrap
 
   include ChartHelper
 
-  def initialize(chart_type:, school:, chart_config: nil, analysis_controls: true, no_zoom: true, axis_controls: true, html_class: 'analysis-chart', fuel_type: nil)
+  def initialize(chart_type:, school:, chart_config: nil, analysis_controls: true, no_zoom: true, axis_controls: true, html_class: 'analysis-chart', fuel_type: nil, show_advice: false, autoload_chart: true, wrap: true)
```

Comment:
Adding the autoload option is a good extension, but I'm not sure about the show_advice and wrap settings.

The "show_advice" option on the chart tag is a holdover from much earlier versions of the analytics. I'm not sure its even used anywhere at the moment, it was originally used to inject some content from the JSON response into the page. I'm not sure its actually used anywhere any more. I'd prefer to try and identify that and then remove the option rather than preserve it.

For the "wrap" option, my goal with this change is to remove the need for the chart wrapper, rather than add it to the component. What I was hoping was that we'd be able to replace use of the wrapper divs (which usually contain content before and after the chart, see e.g. custom titles on the pupil analysis controls, or the analysis controls partial ), with the existing chart component slots and options. I think you can achieve the same thing using those.

Can you revisit and see whether you can achieve the same thing with existing features?

Reply:
looks like show advice is set true in `app/views/admin/analysis/generic_chart_template.html.erb` and potentially in `app/views/schools/charts/show.html.erb` (`!params[:control].present?`)  ok to remove for those two?

will look at removing the wrap option and chart wrapper

Reply:
the wrap & show_advice removed from the component options